homeassistant/components/intent/timers.py

```diff
@@ -0,0 +1,772 @@
+
+    return matching_timers
+
+
+def _get_total_seconds(slots: dict[str, Any]) -> int:
+    """Return the total number of seconds from hours/minutes/seconds slots."""
+    total_seconds = 0
+    if "hours" in slots:
+        total_seconds += 60 * 60 * int(slots["hours"]["value"])
+
+    if "minutes" in slots:
+        total_seconds += 60 * int(slots["minutes"]["value"])
+
+    if "seconds" in slots:
+        total_seconds += int(slots["seconds"]["value"])
+
+    return total_seconds
+
+
+class SetTimerIntentHandler(intent.IntentHandler):
+    """Intent handler for starting a new timer."""
+
+    intent_type = intent.INTENT_SET_TIMER
+    slot_schema = {
+        vol.Required(vol.Any("hours", "minutes", "seconds")): cv.positive_int,
+        vol.Optional("name"): cv.string,
+        vol.Optional("device_id"): cv.string,
+        vol.Optional("data"): dict,
+    }
+
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
+        """Handle the intent."""
+        hass = intent_obj.hass
+        timer_manager: TimerManager = hass.data[TIMER_DATA]
+        slots = self.async_validate_slots(intent_obj.slots)
+
+        device_id: str | None = None
+        if "device_id" in slots:
+            device_id = slots["device_id"]["value"]
+
+        name: str | None = None
+        if "name" in slots:
+            name = slots["name"]["value"]
+
+        hours: int | None = None
+        if "hours" in slots:
+            hours = int(slots["hours"]["value"])
+
+        minutes: int | None = None
+        if "minutes" in slots:
+            minutes = int(slots["minutes"]["value"])
+
+        seconds: int | None = None
+        if "seconds" in slots:
+            seconds = int(slots["seconds"]["value"])
+
+        data: dict[str, Any] | None = None
+        if "data" in slots:
+            data = slots["data"]["value"]
+
+        await timer_manager.start_timer(
+            hours,
+            minutes,
+            seconds,
+            language=intent_obj.language,
+            device_id=device_id,
+            name=name,
+            data=data,
+        )
+
+        return intent_obj.create_response()
+
+
+class CancelTimerIntentHandler(intent.IntentHandler):
+    """Intent handler for cancelling a timer."""
+
+    intent_type = intent.INTENT_CANCEL_TIMER
+    slot_schema = {
+        vol.Any("start_hours", "start_minutes", "start_seconds"): cv.positive_int,
+        vol.Optional("name"): cv.string,
+        vol.Optional("ordinal"): cv.positive_int,
+        vol.Optional("device_id"): cv.string,
+    }
+
+    async def async_handle(self, intent_obj: intent.Intent) -> intent.IntentResponse:
+        """Handle the intent."""
+        hass = intent_obj.hass
+        timer_manager: TimerManager = hass.data[TIMER_DATA]
+        slots = self.async_validate_slots(intent_obj.slots)
+
+        timer = _find_timer(hass, slots)
+        await timer_manager.cancel_timer(timer.id)
+
+        return intent_obj.create_response()
```

Comment:
Same here, attach metadata?

Reply:
I'll save this for a follow-up PR when we think more about the frontend.